src/components/MyAccount/Settings/AccountSettingsUtils.ts

```diff
@@ -0,0 +1,121 @@
+import type { IconNames } from "@nypl/design-system-react-components"
+import type { Patron, SierraCodeName } from "../../../types/myAccountTypes"
+import { notificationPreferenceMap } from "../../../utils/myAccountData"
+import { filteredPickupLocations } from "../../../../__test__/fixtures/processedMyAccountData"
+
+type PutRequestPayloadType = {
+  emails?: string[]
+  phones?: Phone[]
+  fixedFields?: {
+    /* eslint-disable @typescript-eslint/naming-convention */
+    268: {
+      label: "Notice Preference"
+      value: string
+    }
+  }
+  homeLibrary?: string
+}
+type Phone = { number: string; type: string }
+type PhoneOrEmail = string | Phone
+
+export const accountSettings = [
+  {
+    field: "phones",
+    icon: "communicationCall",
+    term: "Phone",
+    description: (value: Phone[]) => value[0]?.number,
+  },
+  {
+    field: "emails",
+    icon: "communicationEmail",
+    term: "Email",
+    description: (value: string[]) => value[0],
+  },
+  {
+    field: "notificationPreference",
+    icon: "communicationChatBubble",
+    term: "Notification preference",
+  },
+  {
+    field: "homeLibrary",
+    icon: "actionHome",
+    term: "Home library",
+    description: (value: SierraCodeName) => value?.name,
+  },
+  {
+    field: "pin",
+    icon: "actionLockClosed",
+    term: "Pin/Password",
+    description: (_) => "****",
+  },
+] as {
+  field: string
+  icon: IconNames
+  term: string
+  description?: (any) => string
+}[]
```

Comment:
I think you can create a new type that extends `IconListElementPropType`.

Reply:
There might be a better way to do this, but I actually can't make those the same type. i'm calling this property description because it can be a method to return the description, but the IconListElement type expects a string or a react node, and combining these two types makes issues with that discrepancies. Not sure exactly how to address this. this account settings object doesn't really need a type is maybe one solution?